topic-controller/README.md

```diff
@@ -17,71 +17,61 @@ And also in the other direction:
 
-The private copy allows the controller to usually cope with scenarios where the topic 
+For example, suppose the topic controller is not running, and a ConfigMap "my-topic" gets created. 
+When the controller starts it will lack a private copy of "my-topic", 
+so it can infer that the ConfigMap has been created since it was last running. 
+The controller will create the topic corresponding to "my-topic" and also store a private copy of the 
+metadata for "my-topic".
+
+The private copy allows the controller to cope with scenarios where the topic 
 config gets changed both in Kafka and in Kubernetes/OpenShift, so long as the 
-changes are not incompatible (e.g. both changeing the same topic config key, but to 
+changes are not incompatible (e.g. both changing the same topic config key, but to 
 different values). 
+In the case of incompatible changes, the Kafka configuration wins, and the ConfigMap will 
+be updated to reflect that.
 
 
-## Controller configuration
-
-The controller is configured via a ConfigMap within the K8s/OpenShift cluster.
-
-The data of this ConfigMap supports the following keys:
-
-* `kubernetesMasterUrl`, default: `https://localhost:8443`
-* `kafkaBootstrapServers`, default:`localhost:9092`
-* `zookeeperConnect`, default: `localhost:2181`
-* `zookeeperSessionTimeout` a durection for the timeout on the zookeeper session, default: `2 seconds`
-* `fullReconciliationInterval` a duration for the time between full reconciliations, default: `15 minutes`
-* `reassignThrottle`, the throttle to use when topic updates require topic partition reassignment
-* `reassignVerifyInterval` a duration between executions of the `--verify` stage of partition reassignment, default: `2 minutes`  
-
-The controller watches for changes to the config map and reconfigures itself according.
-
-
-## Controller command line and environment
-
-* `--master-url` (or the `CONTROLLER_K8S_URL` env var if absent from the command line) 
-  the URL of the master apiserver in which to find the topic controller's ConfigMap. 
-* `--config-namespace` (or the `CONTROLLER_K8S_NS` env var if absent from the 
-  command line) the namespace within the master apiserver which contains the topic 
-  controller's ConfigMap.
-* `--config-name` (or the `CONTROLLER_K8S_NAME` env var if absent from the 
-  command line) the name of the config map (with the namespace given by `--config-namespace`)
-  which contains the topic 
-  controller's ConfigMap.
-* `--help` command line help, and exit
-* `--help:config` for help on the topic controller's configuration. 
-
+## Controller environment
 
+The controller is configured from environment variables:
 
-## Possible future directions
+* `TC_CM_LABELS` 
+– The Kubernetes label selector used to identify ConfigMaps to be managed by the controller.
+  Default: `strimzi.io/kind=topic`.  
+* `TC_ZK_SESSION_TIMEOUT`
+– The Zookeeper session timeout. For example `10 seconds`. Default: `20 seconds`.
```

Comment:
The way the `Config` class works currently you do have to write `10 seconds` (it uses a regex to extract an integer quantity and a `TimeUnit`, which is looked up case insensitively). I considered doing it the Kafka way and using `ms` uniformly, but who likes having to get out their calculator to decode ms quantities which aren't familiar?

Reply:
This is my first time I see a "time" configuration specifying the timeunit in a string. To be honest I like the way to use `ms` as unit (and then just an integer as value) but in this case I think that `sec` are enough to avoid huge numbers in the configuration itself.